docs/intro/overrides.rst

```diff
@@ -0,0 +1,349 @@
+       `by default`.
+    2. Whenever one of our declared URL rules matches with a given page URL,
+       then the Page Object associated with that rule `overrides (or replaces)`
+       the default ``GenericProductPage``.
+
+This enables us to conveniently declare which Page Object would be used for a
+given webpage `(based on a page's URL pattern)`.
+
+Let's see this in action by declaring the Overrides in the Page Objects below.
+
+
+Creating Overrides
+------------------
+
+Let's take a look at how the following code is structured:
+
+.. code-block:: python
+
+    from web_poet import handle_urls, ItemWebPage
+
+
+    class GenericProductPage(ItemWebPage):
+        def to_item(self):
+            return {"product-title": self.css("title::text").get()}
+
+
+    @handle_urls("example.com", overrides=GenericProductPage)
+    class ExampleProductPage(ItemWebPage):
+        def to_item(self):
+            ...  # more specific parsing
+
+
+    @handle_urls("anotherexample.com", overrides=GenericProductPage, exclude="/digital-goods/")
+    class AnotherExampleProductPage(ItemWebPage):
+        def to_item(self):
+            ...  # more specific parsing
+
+
+    @handle_urls(["dualexample.com/shop/?product=*", "dualexample.net/store/?pid=*"], overrides=GenericProductPage)
+    class DualExampleProductPage(ItemWebPage):
+        def to_item(self):
+            ...  # more specific parsing
+
+The code above declares that:
+
+    - For sites that match the ``example.com`` pattern, ``ExampleProductPage``
+      would be used instead of ``GenericProductPage``.
+    - The same is true for ``DualExampleProductPage`` where it is used
+      instead of ``GenericProductPage`` for two URL patterns which works as
+      something like:
+
+      - :sub:`(match) https://www.dualexample.com/shop/electronics/?product=123`
+      - :sub:`(match) https://www.dualexample.com/shop/books/paperback/?product=849`
+      - :sub:`(NO match) https://www.dualexample.com/on-sale/books/?product=923`
+      - :sub:`(match) https://www.dualexample.net/store/kitchen/?pid=776`
+      - :sub:`(match) https://www.dualexample.net/store/?pid=892`
+      - :sub:`(NO match) https://www.dualexample.net/new-offers/fitness/?pid=892`
+
+    - On the other hand, ``AnotherExampleProductPage`` is only used instead of
+      ``GenericProductPage`` when we're handling pages from ``anotherexample.com``
+      that doesn't contain ``/digital-goods/`` in its URL path.
+
+.. tip::
+
+    The URL patterns declared in the ``@handle_urls`` annotation can still be
+    further customized. You can read some of the specific parameters in the
+    :ref:`API section <api-overrides>` of :func:`web_poet.handle_urls`.
+
+
+Retrieving all available Overrides
+----------------------------------
+
+The :meth:`~.PageObjectRegistry.get_overrides` method from the ``web_poet.default_registry``
+allows discovery and retrieval of  all :class:`~.OverrideRule` from your project.
```

Comment:
I think get_overrides is not doing discovery, and also it doesn't limit rules to a certain project:

```suggestion
allows retrieval of all :class:`~.OverrideRule` in the registry.
```

Reply:
Good point. Updated this on https://github.com/scrapinghub/web-poet/pull/27/commits/37e5ba1a3696f629801478fe78ea1729164f39df